### DISCUSSION TITLE
Dataset

### DISCUSSION THREAD

**Discussion by Arthur-Systems** [DataWhisk/hvac-occupancy-forecasting#7] [Ideas]
Prerequisites
Python Dependencies

pip install pandas scikit-learn lightgbm sqlalchemy python-dotenv snorkel pyyaml
Setup MySQL with Below Tables:

observation_merged
https://drive.google.com/file/d/1K6pVNVejHWA24mbOsSnUPcyCSua7-VUJ/view?usp=drive_link
https://drive.google.com/file/d/110U1Bx9u2Llsz3B-Ej-EpSCw09djT5g8/view?usp=drive_link
user_location_trajectory_FINE_V2
https://drive.google.com/file/d/1-FG71TTOoa7rnLw1bRYoIoU_MjvVkcf8/view?usp=sharing
https://drive.google.com/file/d/1w0QVgz4REOK-g29X7EUpwwEHf2Qobewy/view?usp=sharing
space
https://drive.google.com/file/d/16SiDkgKzsPiCY3wxVQoaterjx8Ni097h/view?usp=sharing
https://drive.google.com/file/d/1Jb7wQLFtox6zLiVKIEb2vpX5FIg41Nrp/view?usp=sharing
space_metadata
https://drive.google.com/file/d/1BMs-LrSIklUd7TG6lZsji3VaSFqnigoS/view?usp=sharing
https://drive.google.com/file/d/1UPmmrGOoeZcwZJj9Ff77HKxHmEOWytkB/view?usp=sharing
space_occupancy
https://drive.google.com/file/d/1dXnjdKWPJGT1qkUgyMJ3RB5t_ANKxMG6/view?usp=sharing
https://drive.google.com/file/d/13hHWL5AMpXzlpUiO7bdoBMQGnJDX40aJ/view?usp=sharing
faculty_data
https://drive.google.com/file/d/1Vjd22qiW5T7CrsCo6qZ6gKuZplQy1-fc/view?usp=sharing
https://drive.google.com/file/d/1H0PNs8nTpkD2jhhKMJl5znBFOfwPOgBL/view?usp=sharing
user_ap_trajectory
https://drive.google.com/file/d/1OIfcgO97KeEMmcimfr9IVjyvApK8IRrC/view?usp=sharing
https://drive.google.com/file/d/1vRVOO5-P0rU7TT12ArErKHb1-9W_Cw05/view?usp=sharing
region_to_coverage
https://drive.google.com/file/d/1tIC3UBdPajdrpdZ6-WCDMY80YQLcSYvZ/view?usp=sharing
https://drive.google.com/file/d/1swKz10jCTsEHv7Rq-UiWlglRAZB2M9i2/view?usp=sharing
About Files
user_snorkel_functions.py
Reads queries from user_queries.yaml and creates Snorkel Labeling functions from them
Labels users based on labeling functions and stores output in user_test_predictions.csv and user_train_predictions.csv
device_snorkel_functions.py
Reads queries from device_queries.yaml and creates Snorkel Labeling functions from them
Labels users based on labeling functions and stores output in device_test_predictions.csv and device_train_predictions.csv
group_snorkel_functions.py
Reads queries from group_queries.yaml and creates Snorkel Labeling functions from them
Labels users based on labeling functions and stores output in group_test_predictions.csv and group_train_predictions.csv
feature_extraction.py
Extracts feature information using queries from feature_queries.yaml for each user from the database.
train_lightgbm_from_snokel.py
Reads user labels and probability from {}_train_predictions.csv and {}_test_predictions.csv files and chooses users with above inputted threshold as example
Reads all user feature information from user_features
uses LightGBM model to train on this data to classify the rest of data points
stores those classifications and associated probabilities in device_full_predictions.csv, user_full_predictions.csv and group_full_predictions.csv
Model encoders are saved to lgbm_device_classifier.pkl, lgbm_group_classifier.pkl, and lgbm_user_classifier.pkl
user_queries.yaml, device_queries.yaml, group_queries.yaml
Contain the queries where each query becomes a labeling function
Each query has a label associated with it. (group_queries have no set label because label is based on result of query so that is handeled in group_snorkel_functions.py)
Index is always user or payload (based on query) which will be used to get individual user data in Snorkel Labeling Functions
Columns field is the list of columns that the query returns or you can leave it empty if you don't need to store other column information
feature_queries.yaml
This file includes all the queries related to the features needed for LightGBM training
Usage:
Fill out .env file with MySQL database information
Edit and Update device_queries.yaml, group_features.yaml, user_queries.yaml with labeling function queries
Run Snorkel Scripts
python device_snorkel_functions.py
python group_snorkel_functions.py
python device_snorkel_functions.py
Edit and Update feature_queries.yaml with the features LightGBM will train on
Run feature extraction script on all users:
python feature_extraction.py
Run LightGBM Model
python train_lightgbm_from_snorkel.py
docker compose up -d docker compose down celery -A celery_app worker --loglevel=info --concurrency=4 celery -A celery_app worker --loglevel=info -Q query_workers,aggregation_workers,index_workers,api_workers -n all_workers@%h

1468544b-cebf-46a2-b1dc-bd3cbf661112

\copy space (space_id, space_name, parent_space_id, building_room) FROM '/home/einargatc/Downloads/space_dump.csv' WITH DELIMITER E'\t' NULL '\N'; \copy space (space_id, space_name, parent_space_id, building_room) FROM '/home/einargatc/Downloads/space_dump.csv' WITH CSV DELIMITER E'\t' NULL '\N'; \copy space_metadata (roomID, typeID, type, building_room) FROM '/home/einargatc/Downloads/space_metadata_dump.csv' WITH CSV DELIMITER E'\t' NULL '\N'; \copy faculty_data (name, office) FROM '/home/einargatc/Downloads/faculty_data_dump.csv' WITH CSV DELIMITER E'\t' NULL '\N';

32490c1a-6fbf-420d-8461-ecf17054d755

Start Test Server docker compose -f docker-compose.test.yml up --build docker compose up -d

Stop Server docker compose -f docker-compose.test.yml down --volumes --remove-orphans && docker system prune -f

LOCAL START docker compose -f docker-compose.test.yml up fastapi redis celery-worker uvicorn main:app --host 0.0.0.0 --port 8000 --reload psql -h capstone-server.ics.uci.edu -U snorkel -d snorkeldb

START FASTAPI SERVER ONLY - will update later uvicorn main:app --reload

**Comment by vivaciousvivvy** [comment]
Dummy data for occupancy predictions
[predictions_output.csv](https://github.com/user-attachments/files/25553200/predictions_output.csv)

**Comment by vivaciousvivvy** [comment]
# Script to Copy csv file in postrgreSQL table #

Start by copying the csv into sensoria
`scp CSV_TO_COPY UCINETID@sensoria.ics.uci.edu:csv_files/`

SSH into sensoria and run this command. Make sure that the table exists in the DB before running this command.
`./import_data <table_name> <file_name>`
